src/uwtools/drivers/forecast.py

```diff
@@ -253,5 +246,52 @@ def _mpi_env_variables(self, delimiter: str = " ") -> str:
         }
         return delimiter.join([f"{k}={v}" for k, v in envvars.items()])
 
+    def _prepare_config_files(self, run_directory: Path) -> None:
+        """
+        Collect all the configuration files needed for FV3.
+        """
+        if self._dry_run:
+            for call in ("field_table", "model_configure", "input.nml"):
+                log.info(f"Would prepare: {run_directory}/{call}")
+        else:
+            self.create_field_table(run_directory / "field_table")
+            self.create_model_configure(run_directory / "model_configure")
+            self.create_namelist(run_directory / "input.nml")
+
+    def _run_via_batch_submission(self) -> Tuple[bool, List[str]]:
+        """
+        Prepares and submits a batch script.
+
+        :return: A tuple that contains a boolean of success status of the FV3 run and a list of
+            strings that make up the batch script.
+        """
+        run_directory = self.prepare_directories()
+        batch_script = self.batch_script()
+        batch_lines = ["Batch script:", *str(batch_script).split("\n")]
+        if self._dry_run:
+            return True, batch_lines
+
+        assert self._batch_script is not None
+        outpath = run_directory / self._batch_script
+
+        batch_script.dump(outpath)
+        return self.scheduler.submit_job(outpath), batch_lines
+
+    def _run_via_local_execution(self) -> Tuple[bool, List[str]]:
+        """
+        Collects the necessary MPI environment variables in order to construct full run command,
+        then executes said command.
+
+        :return: A tuple containing a boolean of the success status of the FV3 run and a list of
```

Comment:
**This** one is accurate about the success status, because there's no batch system involved: It's running FV3 directly and immediately, and so would know about it's status.